src/VisualStudio/Core/Impl/CodeModel/FileCodeModel.cs

```diff
@@ -177,7 +177,13 @@ internal DocumentId GetDocumentId()
 
         internal void UpdateCodeElementNodeKey(AbstractKeyedCodeElement keyedElement, SyntaxNodeKey oldNodeKey, SyntaxNodeKey newNodeKey)
         {
-            var codeElement = _codeElementTable.Remove(oldNodeKey);
+            EnvDTE.CodeElement codeElement;
+            if (!_codeElementTable.TryGetValue(oldNodeKey, out codeElement))
+            {
+                throw new InvalidOperationException($"Could not find {oldNodeKey} in Code Model element table.");
+            }
+
+            _codeElementTable.Remove(oldNodeKey);
```

Comment:
I think it's too early and I'm missing the key difference here.  But I can't on the surface see a difference between the old code (which checked for a `null` return from `_codeElementTable.Remove`) and the new code which uses `TryGetValue`.